Wikipedia/Code/WMFRandomSectionController.m

```diff
@@ -112,6 +112,9 @@ - (void)getNewRandomArticle {
     if (self.fetcher.isFetching) {
         return;
     }
+
+    [self.delegate controllerWillGetItems:self];
```

Comment:
you can just call self.delegate
`- (void)controller:(id<WMFHomeSectionController>)controller didSetItems:(NSArray*)items`

You do't really need the extra method.

Then on line 103 add:

```
if(self.isFetching){
//add blur
}else{
//remove blur
}
```